bundles/org.eclipse.swt/Eclipse SWT PI/gtk/org/eclipse/swt/internal/gtk4/GTK4.java

```diff
@@ -191,7 +191,29 @@ public class GTK4 {
 	 * @param error cast=(GError **)
 	 */
 	public static final native boolean gtk_file_chooser_set_file(long chooser, long file, long error);
-
+	
+	/* GtkFileDialog */
+	public static final native long gtk_file_dialog_new();
+	/**
+	 * @param self cast=(GtkFileDialog *)
+	 * @param parent cast=(GtkWindow *)
+	 * @param cancellable cast=(GCancellable *)
+	 * @param callback cast=(GAsyncReadyCallback)
+	 * @param user_data cast=(gpointer)
+	 */
+	public static final native void gtk_file_dialog_select_folder(long self, long parent, long cancellable, long callback, long user_data);
+	/**
+	 * @param self cast=(GtkFileDialog *)
+	 * @param result cast=(GAsyncResult *)
+	 * @param error cast=(GError *)
```

Comment:
Should be '(GError **)'. What you have here fails with gtk4.c: In function ‘Java_org_eclipse_swt_internal_gtk4_GTK4_gtk_1file_1dialog_1select_1folder_1finish’:
gtk4.c:737:103: error: passing argument 3 of ‘gtk_file_dialog_select_folder_finish’ from incompatible pointer type [-Wincompatible-pointer-types]
  737 |         rc = (jlong)gtk_file_dialog_select_folder_finish((GtkFileDialog *)arg0, (GAsyncResult *)arg1, (GError *)lparg2);
      |                                                                                                       ^~~~~~~~~~~~~~~~
      |                                                                                                       |
      |                                                                                                       GError * {aka struct _GError *}
In file included from /usr/include/gtk-4.0/gtk/gtk.h:134,
                 from gtk4.h:23,
                 from gtk4_structs.h:18,
                 from gtk4.c:19:
/usr/include/gtk-4.0/gtk/gtkfiledialog.h:118:77: note: expected ‘GError **’ {aka ‘struct _GError **’} but argument is of type ‘GError *’ {aka ‘struct _GError *’}
  118 |                                                       GError              **error);
      |                                                       ~~~~~~~~~~~~~~~~~~~~~~^~~~~

Reply:
``` * @param error cast=(GError *)```.suggestion